VotingApplication/VotingApplication.Web/Api/Controllers/ManageController.cs

```diff
@@ -41,5 +53,94 @@ public ManagePollRequestResponseModel Get(Guid manageId)
         }
 
         #endregion
+
+        #region Put
+
+        public void Put(Guid manageId, ManagePollUpdateRequest updateRequest)
+        {
+
+            #region Input Validation
+
+            if (updateRequest == null)
+            {
+                this.ThrowError(HttpStatusCode.BadRequest);
+            }
+
+            if (updateRequest.Expires && updateRequest.ExpiryDate < DateTime.Now)
+            {
+                ModelState.AddModelError("ExpiryDate", "Invalid or unspecified ExpiryDate");
+            }
+
+            if (!ModelState.IsValid)
+            {
+                this.ThrowError(HttpStatusCode.BadRequest, ModelState);
+            }
+
+            #endregion
+
+            using (var context = _contextFactory.CreateContext())
+            {
+                Poll existingPoll = context.Polls.Where(p => p.ManageId == manageId).Include(p => p.Options).SingleOrDefault();
+
+                if (existingPoll == null)
+                {
+                    this.ThrowError(HttpStatusCode.NotFound, string.Format("Poll for manage id {0} not found", manageId));
+                }
+
+                existingPoll.AnonymousVoting = updateRequest.AnonymousVoting;
+                existingPoll.Expires = updateRequest.Expires;
+                existingPoll.ExpiryDate = updateRequest.ExpiryDate;
+                existingPoll.InviteOnly = updateRequest.InviteOnly;
+                existingPoll.MaxPerVote = updateRequest.MaxPerVote;
+                existingPoll.MaxPoints = updateRequest.MaxPoints;
+                existingPoll.Name = updateRequest.Name;
+                existingPoll.OptionAdding = updateRequest.OptionAdding;
+                existingPoll.RequireAuth = updateRequest.RequireAuth;
+
+                List<Option> newOptions = new List<Option>();
+                List<Option> oldOptions = new List<Option>();
+                List<Vote> oldVotes = new List<Vote>();
+
+                foreach (Option option in existingPoll.Options)
```

Comment:
It's not immediately obvious what this foreach loop is doing. Could it be extracted to a method with a more explanatory name?